tests/test_hamiltonians.py

```diff
@@ -169,6 +177,11 @@ def test_hamiltonian_matmul(backend, sparse_type):
         backend.assert_allclose((H1 @ H2).matrix, (m1 @ m2))
         backend.assert_allclose((H2 @ H1).matrix, (m2 @ m1))
 
+    try:
+        H1 @ np.zeros(3 * (2**nqubits,), dtype=m1.dtype)
+    except Exception as error:
+        print(error)
```

Comment:
Same here, better to handle the exception properly.

Reply:
sorry, it is a leftover from previous tests